-------------------------------
title: Lecture1
date: 06-12-2015
-------------------------------

## Specific Epigenetic marks

### Histone Modification
- Chemical modification: Acetylation of histone tails
- Ubiquination, Phosphorylation, sumoylation, ADP-ribsolyation, citrullination

- Tend to occur at N terminal sites, predominantly on H3,H4
- Can occur on Lysines(K) or Arginine(R): Function transcription or dna repair
- Functional outcome of various permutation/combination possible for modification is called
the “histone code”
- 

## histone Acetylation
- Acetylation leads to gene activity
- Reduces positive charge on histones, thus reducing DNA-histone interaction
- Acts as docking site for other proteins that are chromatin openers
- Acetylation happens through HAT(Histone acetyltransferase)
- Removal by histone deacetyltransferase(HDAC)
- HAT or KAT(Lysine): 18 different versions in mouse
- HDACs - 18 different versions
- Arguabley just called a chromatin modification and not “epigenetic modification”:
    - Due to lack of rapid acetylation/deacetylation dynamics(Can even follow our circadian rhythm for example)
    - Lack of mechansim for mitotic Heritability

## Histone Methylation
- No change in charge so can lead to increased/decreased activity
- H3K4Me  : Active, around the promoter
- H3K9Me  : Inactive, Spread over the gene, Constitutive Heterochromatin
- H3K27Me : Inactive, Spread over the gene,  Facultative Heterochromatin
- H3K9 ladi down by specific version of  DMTs, KMT, removed by KDMs
- Example of H3K9 KMT is Ezh2, part of PRC2

## Influence of histone modification

- Modifed histone tails read by other chromatin proteins
- interacting proteins have the ability to alter chromatin packaging
- Interacting proteins can also bring about other histone modifications
- Acetylated Lysine: Bound by Bromo domain
- Methylated Lysine: Chromo, MBT, PHD, Tutor 
- Phospohorylated Serine: 14-3-3

## Docking with chromatin proteins
- H3K4Me => CHD1 (ATP dependent chromatin remodeller)
- H3K9Me => HP1 (essential chromatin remodeller, can recruit DNMT1)
- CBX2 => Part of polycomb repressive complex that lays down H2AK119ub
- HP1 binds H3K9Me3 => recruit DNMT1 . HP1 can recruit HMT to maintain and spread H3K9Me3
mark. DNMT recruits HDAC


## Summary so far
- Euchromatin: Acetylation
- heterochromatin: H3K(, HeK4, H3K27

## Chromatin remodelling
- Remodellers have th ability to hydrolse ATP to allo remodelling
- Each has a specific binding domain and ATPase
- Almost all vital for mammals
- Tissue specific deletion demonstrates important roles in differentiation

## Remodeller types
- SWI-SNF(SWItch/Sucrose Non Fermentable)
- ISWI( Imitation SWI)
- CHD(Chromo domain and histone like domain)


Histone modification and nucleosome repositioning can be simultaenous or sequential


## Active regions
Unmethyated DNA + Histone acetylation + H3K4Me

## Inactive regions
Methylated DNA + Histone Methylation - H3K9Me, H3K27Me + Histone deacetylation



## Histone Variants
- Different variants for H2A, H3 and H1
- Can be incorporated in DNA replication dependent or independent manner
- Histone variants are deposited for 
    - increased stability v/s decreased stability
    - Amino acids such as Serine can be phosphorlyated

## Histone Variants - H3
- CENP-A forms at the centromere

## H2A.x
- Highly conserved histone variant
- C terminal motif differs from normal H3
- Has Serince at 139, prone to Phosphorylation
- S139 phosphorylated during DSB
- gamma-H2A.x recruits dNA repair proteins, including epigenetic factors that alter 
chromatin state at DSB
- Phosphate cleaves phosphate group

## Macro H2A
- Associated with Inactive X

## Summary
- Structural Aspects: CENP-A
- DNA Replari: H2A.x
-   Transcription silencing: Macro2 H2A


## Non Coding RNAs

### miRNAs Post transcriptional gene silencing
- Synthesis: DNA => pri-mRNA(Stem loop) recognised by DROSHA enzyme => Processed by Dicer
- Following DICER action : asymmetric RISC assmbley(miR or siRNA duplex) -> Two ways
    - Most plants: perfet homology >> mRNA cleavage
    - Partial homology: Translational repression

- Used in RNAi experiments for post transcriptional gene silencing

## piwi-interacting RNAs
- Silence transposable elements in germ line and stem cell population
- 24-35nt in length( microRNAs are 19-24bp)
- Transcribed from uni-directional and bi-directional active transposons or other
repeats 
- Bounded by piwi proteins
- Synthesis: Transposons => piRNA precursors => primary shortening => Loading onto various Piwi
proteins -> Post transcriptional gene silencing
- Primary processing: involves shortening, Secondary processing: Ping-pong amplification

piRNA - piwi protein allow 
- Degradation of transposons RNA: post transcriptional silencing
- DNA methylation of transposon/repetitive element DNA(transcriptional silencing via RNA directed DNA methylation)

## piRNA v/s miRNA
- Both DICER dependent
- miRNA can act only in post transcriptional silencing
- piRNAs can act in transcriptional silencing and can alter epigenetic state

## Importance of piRNA
- important in germ line and in stem cell
- Mutations in PIWI leads to infertility
- Silencing in repetitive regions

## Silencing in non-repetitive regions
- Rasgrf1 : paternally imprinted gene

## Long non coding RNAs
- More than 200bp in length
- usually spliced, capped and polydeanylated
- Constrained in the nucleus
- Are able to regulate epigenetic processes
- Examples
    - X inactivation(Xist)
    - Genomic imprinting
    - Hox gene silencing
    - DNA damage response

## Targeting tp specific sites?
- Most complexes don’t have dna binding capability
- Specificity does not allow for unique targeting
- By binding to specific chromatin marks, still insufficient specificity

- Long noncoding RNAs direct epigenetic machinery in cis or trans
- long noncoding rnas are allele specific as they are tethered to the locus 
in cis during transcription
- Length of molecule and sequence provide unparallel site-specificity

## lnRNAs possible mechanisms:
- Guides(e.g Xist): Guide the epigenetic machinery towards sites
- Scaffolds: Combine two different complexes along with the epigenetic machinery(##NOT Sure about this)
- DEcoys
- Signals(transcription is required, so lnRNAs may be a byproduct)
- Enhancer
- Reservoir

## Xist and Hotair
- Function in X inactivation
- Epigenetic dosage compensation mechanism in mammals to ensure males and females
- Requires allele-specificity as ONLY one of the two Xshould be inactivated
- Xist if EXPRESSED => That X is inactive
- Xist RNA coats the inactive X in cis(the other one)

## Xist mechansism
- Xist contains repeats
- Repeat A is required for silencing function of Xist
- RepA binds the histone methyltrasnferases


## Xist
- acts in cis(transcription of Xist required to establish silencing on same chromosome)
- Acts in sequence specific way
- Recruits chromatin modifier complex(PRC2)


## Hotair
- Cluster on Chr2,12 on mammals
- Acts in trans
- Expressed from HOXC locus, represses the HOXD locus
- Binds to PRC2 and LSD1(H3K4 Demethtylase)
    - PRC2 adds repressive H3K27Me
    - LSD1 removes active H3K4Me

 - Effectively leads to gene silencing


## 3D organisation of the nucleus

- Nuclear lamnina: Polycomb body
- Nuclear pore: Paraspeckles
- Nucleolus: Splicing bodies
- Transcription Factories



## Nuclear periphary:

- Associated with silencing
- Associated with facultative heterochromatin
- Inner nuclear compartments: euchromatic
- REgions attached to the lamina are lowly expressed]]
- Possibly activation should involve moving themachinery away from the periphary

## Transcription factories

- Concentrated regions of RNA Pol ii 
- RNA Pl ii anchored, DNA passes through poll ii
- Coregulated genes are localised to the same factory

## Nuclear pores
- Localisation to the nuclear pore-euchromatic, actively Transcribed
- Fast export to cytoplasm for translation?


## Nucleolus
- Aggregation of ribosomal DNA repeats from different chromosomes
- Site of rRNA transcription by RNA pol i
- Localisation important for functioning
## Peri nucleolar space
- Low RNA pol ii and generally repressed
- Exception - tRNA genes, Rna pol iii transcribed 

### Summary!

- Active marks: HeK4Me, Acetylation. The chromatin remodelling complexes
cause the histone marks to be sparsely located

- Inactive: H3K9Me, H3K27Me, DNA methylation, nucleaosome are densely packed

- 3pigenetic marks act as docking marks for other proteins

- piRNAS direct methylation of transposable elements

- lnRNAs direct epigentic machinery to particular sites


## Heritability of epigenetic marks
- DNA methylation is mitotically heritable
- Histone methylation can be heritable:
    - Like for H3K27Me by PRC2
- Histone acetylation not heritable
    - Changes too rapidly to be heritable
- histone variants can be heritable
- LnRNAs can be heritable
    - Xist!
- Effect of piRNAs can be heritable
    - Setup a cascade of events
    - Such as dna methylation which is heritable
-Nuclear architecture is possibly heritable


