-------------------------------
title: Lecture1
date: 06-05-2015
-------------------------------

##Epigenetic Control
- Epigenetics: Extra information above the genetic information
that must be present to enable differentiation and development
- Different set of genes are expressd in different cell types.
E.g Nueron: Dopamine, RBC: HB, Muscle: Myoglobin
- Different set of genes expression driven by: TF binding specificity + Epigenetics
- Epigenetic marks demarcate start and end of gene, provie structure to the chromosome
leads to gene being expressed or repressed


##Epigenetic throughout development 
- Zygote(Fertilised Egg) => Preimplantation embryo(Before being planted to
Uterus to make placenta) => Postimplantation embryo => Midgestation embryo
=> Blood cell development

- Identical Twins : Derived from the same egg and same sperm 
so genetically identical. But epigenetically different
- 

-Epigenetics(Art Riggs et al, 1996): is the study of mitotically
heritable changes in gene expression that happen without changes in
DNA sequence

- Mitotical heritability: Same set of cells expressed in daughter cells
(tissue homegenous). Lack of heritability will give rise
to heteogenous tissues


- Epigenetic repgrogamming happens in germ cells and early development

## Chromatin structure and epigenetics

- DNA+histones = chromatin
- 2 meters of DNA compacted by chromatin to ~10 mu

- Compact DNA creates problems for accessibility during transcription
and replication

-Tightly packed: DNA Less accessible and hence Transcription silent

- Nucleosome = DNA + histones
	- 146 bp of DNA wrapped around octamer(2x): H2A,H2B,H3, H4 linked by H1
	- DNA makes 1.75 turns about the octamer
	- Histones(lysine and argine rich) are + charged and DNA is - charged
	- N terminals protrude from the histones
	- Nucleosomes + interlinked DNA = beads on string


## DNA packaging into chromatin

- Nucleosome chromatin further packaged into 30nm fibre, dependent on H1 interactions
- 30 nm fibre attaches to scaffold proteins to further condense chromosome
- During interphase(No dividing stage): the chromosomes are packed as 30nm fibres
- Metaphase chromsome: More scaffold proteins -> required for segregation of chromosomes
during cell division
-  

## Hetero vs Euchromatin
- Heterochromatin: Closed, densely packed
	-Facultative: Can differ by cell type
	-Constitutive: Same in all cell types
		- Constitutes a structural role
		- Found at telomeres, centromeres, sex chromosomes(mostly Y)
- Metaphase chromatin differs for different mammals.

- Euchromatin: Open, less dense


- Heterochromatin function:
	- Gene silencing
	- Maintain the integrity of the genome


## Specific DNA marks

- DNA Methylation:
	- Additing on CH3 tat CpG dinucleotides, happens at Cytosine(5')
	- CpG are symmetrical (C-phophate-G): Allows them to be maintained during cell division
	- In mammals, DNA methylation laid down by DNMT3a and DNMT3b(de-novo methylation)
	- Conside rtwo strands: --C_mG--+--GC_m-- => (--C_mG-- + --GC--) ++ (--GC_m-- + --CG--)
	The last two are hemi-methylted(1 strand methylated, other not)
	- DNMT1 maintains methylation
	- Methlylationat CpG = Gene silenced
	- CpG clustered as CpG islands, often at promoters, the site of TF binding
	- CpG islands tend to be protected from methylation

- X inactivatio
	- Mitotic heritability of seX chromosomes
	- Dosage compensation(Female and males have same compsn of X chr)
	- Random X inactivation happens during gastrulation in embryo => inherited by all! daughter cells(The X Either from your mom or dad)
	- CpG islands ARE methylated in inactive X(Generally islands are void of any methylation)
	- Calico cats are lovely because they undergo random X inactivation
	
- How Silencing?
	- me_CpG are associated with chromatin condensation (repressive chromatin structure or 1 degree mechanism)
	- me_CpG bound by MECP1,2 proteins
	- MeCP proteins -> DNA binding + Transcription repression domain
	- MeCP can also recruit others 
	- me_CpG can also prohibit TF binding (2nd degree mechanism) [Very rare]

- Where?
	- CpG:  UNmethylated
	- Intergenic: MEthylated
	- Repeats: MEthylated
	- CpG are under represented as methylated C is prone to deamination(C->T)[mutagenic]

- Why intergenic?
	- Maintain genomic stability, integrity
	- Dnmt1(methylation maintainer) null cells dispaly genomic instability
	- Abnormal karyotype from methylation : Deletion, insertion, relocation
	- Also silences cryptic splice sites
- Why ME at repeats?
	- Maintain integrity
	- IAP = LTR1 -- GAG Protein -- Pol Protein -- LTR2: Really really required
		- Prevent mutageninc transposition
		- Gag and Pol needs to be active
		- If LTR2 active: Downstream expressed: NOT required, need to methykate CpG on LTR2
		- mutation of repeats to prevent transposition
		- Avoids transcriptional interference from strong promoters
        -   Prevent illegitimate recombination

- Methlylationa and Cancer
    - Historically, the first one found was hypomethylation at intergenic regions
    - More consistent thatn just mutation profile
    - DNA demethylation occurs in early development, in primoridual germ cell development
    and later at specific stages of differentiation
    - Indepdent of DNA replication(indepedent of  DNMT1) => Active demethylation

- Passive demethylation
    - Dilution of DNA methylation with cell division when DNMT1 is absent

- Active demethylation
    - C-C is difficult to remove so it must be the enzymes that remove the C-C group through a lot of intermediates

## Summary
-   Occurs at CpG dinucleotides
-  Gene silencer
- Genomic stability
- DNMT1 for maintaining, 3a3b for de-novo
- Mitotically heritable due to DNMT1 maintainer
- Can be reoved actively/passively involving TET proteins
- Essential for viability [DNMT1 null leads to death]

## Heritability of DNA Methylation
- DNMT1 recognises hemi-methylated DNA: restores methylation on relicated strand
- TET proteins, rarely expressed, are involved in dna demethylation

